app/src/main/java/com/yucwang/turtle/Services/TurtleService.kt

```diff
@@ -0,0 +1,128 @@
+package com.yucwang.turtle.Services
+
+import android.app.*
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.os.AsyncTask
+import android.os.Binder
+import android.os.IBinder
+import com.yucwang.turtle.Backend.AppUsageManager
+import com.yucwang.turtle.Overview.HistoryListDatabase
+import com.yucwang.turtle.Overview.OverviewHistoryListItem
+import com.yucwang.turtle.R
+import java.util.*
+
+/**
+ * The turtle class, used to monitor the current usage of android devices
+ */
+class TurtleService : Service() {
+    companion object {
+        val INVOKER_NAME = "invoker_name"
+        val NORMAL_BROADCAST = "normal_broadcast_sync"
```

Comment:
We will also send information when the app is on, we'd better also add an invoker for this now.